### ISSUE TITLE
New Windows x86 failure: `testMemoryErrorBigSource` in `test_exceptions`

### ISSUE THREAD

**Issue by sobolevn** [python/cpython#110767]
# Bug report

```
 ======================================================================
FAIL: testMemoryErrorBigSource (test.test_exceptions.ExceptionTests.testMemoryErrorBigSource)
----------------------------------------------------------------------
OverflowError: cannot fit 'int' into an index-sized integer

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\a\cpython\cpython\Lib\test\test_exceptions.py", line 322, in testMemoryErrorBigSource
    with self.assertRaisesRegex(OverflowError, "column offset overflow"):
AssertionError: "column offset overflow" does not match "cannot fit 'int' into an index-sized integer"

----------------------------------------------------------------------
Ran 1 test in 0.010s

FAILED (failures=1)
```

https://github.com/python/cpython/actions/runs/6494069893/job/17636283375

**Comment by lysnikolaou** [comment]
Related to #110754. Pushed fix on #110768.

**Comment by lysnikolaou** [comment]
Fixed in https://github.com/python/cpython/pull/110768.
